pkg/operator/controllers/checkers/ingresscertificatechecker/checker.go

```diff
@@ -59,13 +62,35 @@ func (r *checker) Check(ctx context.Context) error {
 		return err
 	}
 
-	if ingress.Spec.DefaultCertificate == nil {
-		return errNoDefaultCertificate
+	cluster, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	return validateCertificate(cv.Spec.ClusterID, ingress.Spec.DefaultCertificate, cluster.Spec.Domain)
+}
+
+func validateCertificate(clusterId configv1.ClusterID, ingressCertificate *corev1.LocalObjectReference, clusterDomain string) error {
+	isManagedDomain := dns.IsManagedDomain(clusterDomain)
+
+	if ingressCertificate == nil {
+		if isManagedDomain {
+			return errNoCertificateAndManagedDomain
+		}
+		return errNoCertificateAndCustomDomain
 	}
 
-	if ingress.Spec.DefaultCertificate.Name != string(cv.Spec.ClusterID)+"-ingress" {
-		return fmt.Errorf("custom ingress certificate in use: %q", ingress.Spec.DefaultCertificate.Name)
+	certificateName := ingressCertificate.Name
+	isCertificateNameValid := isCertificateNameValid(certificateName, clusterId)
+
+	if !isCertificateNameValid && isManagedDomain {
```

Comment:
```suggestion
	if !isCertificateNameValid(certificateName, clusterId) && isManagedDomain {
```

Reply:
adding up on this, renaming the function name to a statement instead of a question would probably be good too. `certificateIsValid(...)` is easier to read as a boolean

Reply:
I leave this one on Aldo. We use both in our codebase. I personally do not have strong opinions on the naming.

Reply:
Done. Tks!